packages/ui-kit/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,102 @@
+import { ClickAwayListener } from '@mui/base/ClickAwayListener'
+import { Select as MUISelect, SelectProps as MUISelectProps } from '@mui/base/Select'
+import { SelectOptionDefinition, useSelect } from '@mui/base/useSelect'
+import { prepareForSlot } from '@mui/base/utils'
+import classNames from 'classnames'
+import * as React from 'react'
+import { Button, ButtonProps } from '../Button'
+import { ChevronDownIcon } from '../Icons/ChevronDown'
+import { Option } from './Option'
+import componentStyles from './Select.module.scss'
+
+export interface SelectProps
+  extends Pick<ButtonProps, 'startAdornment' | 'endAdornment' | 'size'>,
+    MUISelectProps<string, false> {
+  options?: SelectOptionDefinition<string>[]
+}
+
+const ButtonSlot = prepareForSlot(Button)
+
+export const Select = React.forwardRef<HTMLButtonElement, SelectProps>(
+  (
+    {
+      options,
+      startAdornment,
+      endAdornment = () => <ChevronDownIcon size={18} />,
+      value: valueProp = undefined,
+      children,
+      className,
+      slotProps,
+      listboxOpen,
+      size = 'default',
+      onListboxOpenChange,
+      ...rest
+    },
+    forwardedRef
+  ) => {
+    const [listboxVisible, setListboxVisible] = React.useState(listboxOpen ?? false)
+    const { getButtonProps, value, open } = useSelect<string, false>({
+      onOpenChange: setListboxVisible,
+      open: listboxVisible,
+      value: valueProp,
+      ...rest
+    })
+
+    React.useEffect(() => {
+      if (onListboxOpenChange) {
+        onListboxOpenChange(listboxVisible)
+      }
+    }, [onListboxOpenChange, listboxVisible])
+
+    React.useEffect(() => {
+      setListboxVisible(listboxOpen ?? false)
+    }, [listboxOpen])
+
+    React.useEffect(() => {
+      setListboxVisible(false)
+    }, [value])
+
+    return (
+      <ClickAwayListener onClickAway={() => setListboxVisible(false)}>
+        <MUISelect
+          ref={forwardedRef}
+          value={value}
+          className={classNames(componentStyles.rootSelect, className)}
+          {...rest}
+          slots={{ root: ButtonSlot }}
+          slotProps={{
+            root: getButtonProps({
+              startAdornment,
+              endAdornment,
+              size,
+              value: value ?? undefined,
+              className: classNames(componentStyles.button, componentStyles.rootButton, {
+                [componentStyles[size]]: size && size !== 'default'
+              })
+            }),
+            popper: {
+              className: classNames(
+                componentStyles.popper,
+                { [componentStyles[size]]: size && size !== 'default' },
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                slotProps?.popper?.className
```

Comment:
It's not absolutely necessary. It's an optional prop that helps customize popper styles for advanced scenarios like [this one](https://github.com/propeldata/ui-kit/blob/1541a126676c25aca9b2baf610b63e02142de516/packages/ui-kit/src/components/TimeRangePicker/TimeRangePicker.tsx#L271).

Reply:
Sorry, maybe I did not explain correctly, when it's necessary to bypass typescript/eslint checks we should add a comment explaining why it's necessary to do that as this might bring confusion for developers in the future, also, please check if the proposed fix makes sense so we don't need to bypass typescript/eslint at all